uint32 number_of_symbols
string[] symbol_names
